src/main/java/org/galatea/starter/entrypoint/SettlementRestController.java

```diff
@@ -34,34 +37,60 @@
 @RestController
 public class SettlementRestController {
 
-  @NonNull
-  SettlementService settlementService;
-
-  public static final String SETTLE_MISSION_PATH = "/settlementEngine";
-  public static final String GET_MISSION_PATH = SETTLE_MISSION_PATH + "/mission/";
-
-  @PostMapping(value = SETTLE_MISSION_PATH, consumes = {MediaType.APPLICATION_JSON_VALUE})
-  public ResponseEntity<Set<String>> settleAgreement(
-      @RequestBody final List<TradeAgreement> agreements) {
-    Set<Long> missionIds = settlementService.spawnMissions(agreements);
-    Set<String> missionIdUris =
-        missionIds.stream().map(id -> GET_MISSION_PATH + id).collect(Collectors.toSet());
-
-    return ResponseEntity.accepted().body(missionIdUris);
-  }
-
-  @GetMapping(value = GET_MISSION_PATH + "{id}", produces = {MediaType.APPLICATION_JSON_VALUE})
-  public ResponseEntity<SettlementMission> getMission(@PathVariable final Long id) {
-
-    Optional<SettlementMission> msn = settlementService.findMission(id);
-
-    if (msn.isPresent()) {
-      return ResponseEntity.ok(msn.get());
-    }
-
-    return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-  }
-
-
+	@NonNull
+	@Autowired
+	SettlementService settlementService;
+
+	public static final String SETTLE_MISSION_PATH = "/settlementEngine";
+	public static final String GET_MISSION_PATH = SETTLE_MISSION_PATH + "/mission/";
+
+	// @PostMapping to link http POST requests to this method
+	// @RequestBody to have the post request body deserialized into a list of TradeAgreement objects
+	@PostMapping(value = SETTLE_MISSION_PATH, consumes = { MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<RestResponse> settleAgreement(@RequestBody final List<TradeAgreement> agreements,
+			@RequestParam(value = "requestId", required = false) String requestId) {
+
+		// if an external request id was provided, grab it
+		processRequestId(requestId);
```

Comment:
Also, will have to do some more looking into how to use @Log for the JMS endpoint. We're currently using SpringAOP, which only works around Spring managed beans (ref: http://aspect4log.sourceforge.net/quick-start-springaop.html). The FuseMessageListenerContainer is instantiated by the FuseJmsListenerContainerFactory, not spring. Also SpringAOP only allows @Log annotations on public methods, and invokeListener is protected.